bindings/c/test/mako/mako.cpp

```diff
@@ -447,22 +457,26 @@ int runOneTransaction(Transaction& tx,
 		auto f = opTable[op].stepFunction(step)(tx, args, key1, key2, val);
 		auto future_rc = FutureRC::OK;
 		if (f) {
-			if (step_kind != StepKind::ON_ERROR) {
-				future_rc = waitAndHandleError(tx, f, opTable[op].name());
-			} else {
-				future_rc = waitAndHandleForOnError(tx, f, opTable[op].name());
+			if (!waitFuture(f, opTable[op].name())) {
```

Comment:
I think fixing waitAndHandle[...] functions would imply that we pass a `ThreadStatistics` object (possibly as optional argument) to them and then do the updates of the error statistics inside these functions. We do not have access to the initial error outside of the waitAndHandle[...] functions. What made me do it the way I did it was mostly that it made the code more similar to the async case (which I thought is a good thing?). I think passing the ThreadStatistics as an optional argument would work as well, but then we need to ensure that every part of the code that calls `waitAndHandle[...]` wants to update its `ThreadStatistics` object the same way. This would include including an extra header in `future.hpp`